phpBB/phpbb/cron/event/cron_runner_listener.php

```diff
@@ -0,0 +1,101 @@
+<?php
+/**
+ *
+ * This file is part of the phpBB Forum Software package.
+ *
+ * @copyright (c) phpBB Limited <https://www.phpbb.com>
+ * @license GNU General Public License, version 2 (GPL-2.0)
+ *
+ * For full copyright and license information, please see
+ * the docs/CREDITS.txt file.
+ *
+ */
+
+namespace phpbb\cron\event;
+
+use phpbb\cron\manager;
+use phpbb\lock\db;
+use phpbb\request\request_interface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\KernelEvents;
+use Symfony\Component\HttpKernel\Event\PostResponseEvent;
+
+/**
+ * Event listener that executes cron tasks, after the response was served
+ */
+class cron_runner_listener implements EventSubscriberInterface
+{
+	/**
+	 * @var \phpbb\lock\db
+	 */
+	private $cron_lock;
+
+	/**
+	 * @var \phpbb\cron\manager
+	 */
+	private $cron_manager;
+
+	/**
+	 * @var \phpbb\request\request_interface
+	 */
+	private $request;
+
+	/**
+	 * Constructor
+	 *
+	 * @param db 				$lock
+	 * @param manager			$manager
+	 * @param request_interface	$request
+	 */
+	public function __construct(db $lock, manager $manager, request_interface $request)
```

Comment:
instead of injecting the request you could add a state to the event_listener and subscribe to Kernel::Request too (you will have the symfony request but I don't think that's an issue here, rigth?)